sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -608,7 +608,8 @@ private[execution] final class LongToUnsafeRowMap(val mm: TaskMemoryManager, cap
   def optimize(): Unit = {
     val range = maxKey - minKey
```

Comment:
Not a Scala expert so I'll trust your judgement! 

`maxKey` being initialized to `Long.MinValue`, I wasn't sure of all the cases where `range` could end up negative so `range >=0` seemed the safest thing to do.